core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -137,7 +137,9 @@ public static List<String> manifestListLocations(Table table, Set<Long> snapshot
    *
    * @param table table for which statistics files needs to be listed
    * @return the location of statistics files
+   * @deprecated use the {@code allStatisticsFilesLocations(table)} instead.
    */
+  @Deprecated
```

Comment:
removed the changes in this file as the new APIs added was not called from anywhere (this is needed for spark action). 

I will raise a follow up PR for Spark actions to consider partition stats file for remove orphan files and expire snapshots action.

Reply:
Since, it was needed only for spark action, moved these changes into a separate PR with the tests for expire snapshots and remove orphan files. 

https://github.com/apache/iceberg/pull/9284